addons/mass_mailing/models/mail_mail.py

```diff
@@ -64,8 +64,12 @@ def _send_prepare_body(self):
 
             # generate tracking URL
             tracking_url = self._get_tracking_url()
-            if tracking_url:
-                body = tools.append_content_to_html(body, tracking_url, plaintext=False, container_tag='div')
+            body = tools.append_content_to_html(
+                body,
+                '<img src="%s" alt=""/>' % tracking_url,
```

Comment:
- what's the `alt` for?
- was it even working before without the `img` tag?
- not directly related to your change, but `container_tag` is for plaintext, here it doesn't seem to do anything?

Reply:
@odony No, it doesn't work without the img tag, the objective is to make the customer load the image (and so call the URL) when he open the email (so we can know if the mail is opened, even if the user didn't click on a link)

(ok for the other points :) )

Reply:
(the image tag was already there ;) )

Reply:
My bad, I missed the change of return format for `_get_tracking_url()`, sorry ;-)